### PR TITLE
Remove duplication in SwapAmountInRoute

### PR DESCRIPTION
**Pull request by phamminh0811**
Closes: #3921

## What is the purpose of the change

> This PR remove SwapAmountInRoute implementations from gamm module and update new wasm file to fix ibc_middle_ware test 

*(E.g.: This pull request improves documation of area A by adding ....*

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by mattverse** [comment]
whoa what's the reason this PR has so much changes in files? Looks like you were trying to fix the cosmwasm contracts CI, is the CI failing as a cause of some of the changes you made or was it just failing even without your changes? @phamminh0811

**Comment by phamminh0811** [comment]
i think those CI problem cause by me. There's an error show that the code need a new wasm file. So I was trying to reimplement osmosis-std to the contract and recompile because the osmosis-std version 0.13.2 does not have switchrouter module.  
If you need, I'll remove them and only keep wasm files one.

File: proto/osmosis/gamm/v1beta1/tx.proto [hunk 1]
```diff
@@ -75,59 +71,6 @@ message MsgExitPoolResponse {
   ];
 }
 
-// ===================== MsgSwapExactAmountIn
-message SwapAmountInRoute {
-  uint64 pool_id = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
-  string token_out_denom = 2
-      [ (gogoproto.moretags) = "yaml:\"token_out_denom\"" ];
-}
```
Comment (p0mvn) [inline proto/osmosis/gamm/v1beta1/tx.proto:83]: We only need to remove this.

Please do not remove any messages or queries. We need to keep those for backward compatibility for clients.

Replace `SwapAmountInRoute` in gamm with `SwapAmountInRoute` in `poolmanager`

Let me know if anything doesn't make sense

**Review by p0mvn — CHANGES_REQUESTED** [review]

File: proto/osmosis/gamm/v1beta1/tx.proto (hunk 1 above)
Comment (phamminh0811) [reply proto/osmosis/gamm/v1beta1/tx.proto:83]: Got it

**Pull request closed without merging**